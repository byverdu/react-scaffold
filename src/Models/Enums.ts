export enum VisibilityFilterEnum {
  SHOW_ALL = 'SHOW_ALL',
  SHOW_DONE = 'SHOW_DONE',
  SHOW_TODO = 'SHOW_TODO'
}

export enum ApiRoutes {
  getTodos = '/todos',
  addTodo = '/todos/add',
  updateTodo = '/todos/update',
  deleteTodo = '/todos/delete'
}

export enum HTTPVerbs {
  get = 'GET',
  post = 'POST',
  patch = 'PATCH',
  delete = 'DELETE'
}

export enum LoggerTypes {
  success = 'success',
  error = 'error',
  warn = 'warn',
  empty = ''
}
