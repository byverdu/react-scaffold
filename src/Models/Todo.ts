import { Map } from 'immutable';

export interface Todo {
  text: string;
  id: string;
  done: boolean;
}

export interface TodoState {
  todos: Map<string, Todo>;
  todosLength: number;
}

export interface SetTodoStatus {
  id: string;
  status: boolean;
}
