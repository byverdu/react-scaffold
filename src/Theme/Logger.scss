.visible {
  display: block;
}

.hidden {
  display: none;
}

.success {
  background-color: darken(greenyellow, 30);
  color: white;
}

.warn {
  background-color: orange;
  color: antiquewhite;
}
