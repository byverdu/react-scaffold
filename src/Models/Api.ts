import { Todo } from './Todo';
import { Map } from 'immutable';
export interface API {
  get: (url: string) => any;
  post: (url: string, data: any) => any;
  patch: (url: string, data: any) => any;
  deleteItem: (url: string) => any;
}

export interface APIResponsesData {
  addTodo: Todo;
  getTodos: Map<string, Todo>;
}
