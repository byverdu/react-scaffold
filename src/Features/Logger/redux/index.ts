import { LoggerState } from 'Models/Logger';
import { LoggerTypes } from 'Models/Enums';
export const TOGGLE_LOGGER = 'TOGGLE_LOGGER';
export const TOGGLE_LOGGER_VISIBILITY = 'TOGGLE_LOGGER_VISIBILITY';

export const initialState: LoggerState = {
  type: undefined,
  message: '',
  isVisible: false
};

export const toggleLogger = (payload) => ({
  type: TOGGLE_LOGGER,
  payload
});

export const toggleLoggerVisibility = () => ({
  type: TOGGLE_LOGGER_VISIBILITY,
  payload: {
    type: LoggerTypes.empty,
    message: '',
    isVisible: false
  }
});

export const logger = (
  state: LoggerState = initialState,
  { type, payload }
): LoggerState => {
  switch (type) {
    case TOGGLE_LOGGER:
      return {
        ...payload
      };
    case TOGGLE_LOGGER_VISIBILITY:
      return {
        ...payload
      };
    default:
      return state;
  }
};
