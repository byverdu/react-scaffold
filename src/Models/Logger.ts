import { LoggerTypes } from 'Models/Enums';

export interface LoggerState {
  isVisible: boolean;
  message: string;
  type: LoggerTypes;
}
