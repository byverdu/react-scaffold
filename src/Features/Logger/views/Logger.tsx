import React, { useEffect } from 'react';
import {
  LoggerMapStateToProps,
  LoggerMapDispatchToProps
} from 'Features/Logger/views/LoggerContainer';
import 'Theme/Logger.scss';

const classnames = require('classnames');

export interface LoggerProps
  extends LoggerMapStateToProps,
    LoggerMapDispatchToProps {}

const Logger: React.SFC<LoggerProps> = ({
  message,
  type,
  isVisible,
  toggleVisibility
}) => {
  const classNames = classnames(
    'logger',
    { visible: isVisible },
    { hidden: !isVisible },
    type
  );

  useEffect(() => {
    if (isVisible) {
      setTimeout(() => {
        toggleVisibility();
      }, 2000);
    }
  });

  return <div className={classNames}>{message}</div>;
};

export default Logger;
