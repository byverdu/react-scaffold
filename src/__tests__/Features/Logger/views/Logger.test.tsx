import React from 'react';
import ReactDOM from 'react-dom';
import Logger from 'Features/Logger/views/Logger';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LoggerTypes } from 'Models/Enums';

describe('<VisibilityFilter />', () => {
  let wrapper;
  const component = () => (
    <Logger
      message="Logger Component"
      isVisible={true}
      type={LoggerTypes.success}
      toggleVisibility={() => {}}
    />
  );

  beforeEach(() => {
    wrapper = mount(component());
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component(), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should create a snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
