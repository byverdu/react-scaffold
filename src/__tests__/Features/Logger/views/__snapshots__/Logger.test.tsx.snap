// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisibilityFilter /> should create a snapshot 1`] = `
<Logger
  isVisible={true}
  message="Logger Component"
  toggleVisibility={[Function]}
  type="success"
>
  <div
    className="logger visible success"
  >
    Logger Component
  </div>
</Logger>
`;
